components/my-account/User/UserTable/UserTable.tsx

```diff
@@ -0,0 +1,165 @@
+    flex: { md: 1, xs: 0.5 },
+    maxWidth: { xs: '120px', sm: '130px', md: '100%' },
+    overflow: { xs: 'hidden' },
+    textOverflow: { xs: 'ellipsis' },
+  },
+}
+
+const UserTable = (props: UserTableProps) => {
+  const { b2bUsers, onDelete, onSave } = props
+
+  const { t } = useTranslation('common')
+  const { showModal, closeModal } = useModalContext()
+  const theme = useTheme()
+  const mdScreen = useMediaQuery(theme.breakpoints.up('md'))
+  const [editUserId, setEditUserId] = useState<string | undefined>(undefined)
+
+  const onEditUserButtonClick = (b2BUser: B2BUser) => {
+    if (mdScreen) {
+      setEditUserId(b2BUser.userId as string)
+      return
+    }
+    showModal({
+      Component: UserFormDialog,
+      props: {
+        isEditMode: true,
+        isUserFormInDialog: true,
+        formTitle: t('edit-user'),
+        b2BUser,
+        onSave: (b2BUserInput: B2BUserInput) => onSave(b2BUserInput),
+        onClose: () => {
+          setEditUserId(undefined)
+          closeModal()
+        },
+      },
+    })
+  }
+
+  return (
+    <Table>
+      {!b2bUsers?.length ? (
+        <caption style={{ textAlign: 'center' }}>{t('no-record-found')}</caption>
+      ) : null}
+      <TableHead>
+        <TableRow style={{ backgroundColor: theme.palette.grey[100] }}>
+          <TableCell
+            colSpan={2}
+            sx={{ flex: 1, width: { xs: '150px' }, overflow: { xs: 'hidden' } }}
+          >
+            {t('email')}
+          </TableCell>
+          {mdScreen && (
+            <>
+              <TableCell>{t('first-name')}</TableCell>
+              <TableCell>{t('last-name-or-sur-name')}</TableCell>
+            </>
+          )}
+          <TableCell>{t('role')}</TableCell>
+          {mdScreen && <TableCell>{t('status')}</TableCell>}
+          <TableCell></TableCell>
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        {b2bUsers?.map((b2bUser) =>
+          editUserId && editUserId === b2bUser?.userId ? (
+            <TableRow key={b2bUser?.userId}>
+              <TableCell colSpan={7} style={{ width: '100%', padding: 0 }}>
+                <UserForm
+                  isEditMode={true}
+                  b2BUser={b2bUser}
+                  onClose={() => setEditUserId(undefined)}
+                  onSave={onSave}
+                />
+              </TableCell>
+            </TableRow>
+          ) : (
+            <TableRow key={b2bUser?.userId}>
+              <TableCell colSpan={2} sx={style.emailAddressCell}>
+                {b2bUser?.emailAddress}
+              </TableCell>
+              {mdScreen && (
+                <>
+                  <TableCell sx={{ flex: 1 }}>{b2bUser?.firstName}</TableCell>
+                  <TableCell sx={{ flex: 1 }}>{b2bUser?.lastName}</TableCell>
+                </>
+              )}
+              <TableCell sx={{ flex: 1 }}>
+                {(b2bUser?.roles && b2bUser?.roles.length && b2bUser?.roles[0]?.roleName) || 'N/A'}
```

Comment:
Can't we get these values from getters ?

Reply:
I have changed 
b2bUser?.roles && b2bUser?.roles.length && b2bUser?.roles[0]?.roleName

to 

b2bUser?.roles?.[0]?.roleName || 'N/A'

which is more readable.